src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -127,33 +124,81 @@ public IList<VSTestResult> GetVSTestResults(XmlNode resultNode)
         /// </summary>
         private TestCase MakeTestCaseFromXmlNode(XmlNode testNode)
         {
+            var methodName = testNode.GetAttribute("methodname");
+            var displayName = testNode.GetAttribute("name");
+            var className = testNode.GetAttribute("classname");
+
             var testCase = new TestCase(
                                      testNode.GetAttribute("fullname"),
                                      new Uri(NUnitTestAdapter.ExecutorUri),
                                      _sourceAssembly)
             {
-                DisplayName = testNode.GetAttribute("name"),
+                DisplayName = displayName,
                 CodeFilePath = null,
                 LineNumber = 0
             };
 
             if (_collectSourceInformation && _navigationDataProvider != null)
             {
-                var className = testNode.GetAttribute("classname");
-                var methodName = testNode.GetAttribute("methodname");
                 var navData = _navigationDataProvider.GetNavigationData(className, methodName);
                 if (navData.IsValid)
                 {
                     testCase.CodeFilePath = navData.FilePath;
                     testCase.LineNumber = navData.LineNumber;
+                }               
+            }
+            else
+            {
+                bool isTestNameScenario = IsTestNameScenario(methodName, displayName);
+
+                if (isTestNameScenario)
+                {
+                    // Stash property in test case object
+                    testCase.SetPropertyValue(Constants.TestCaseAdjustedFQNProperty, string.Format("{0}.{1}", className, methodName));
```

Comment:
Why do we give you two things in one property rather than setting two different properties? It seems to me that giving you __individual__ properties for whatever information you need is a much better long term strategy. Then you can make use of them in any way you like.

This is important because __we will have to maintain whatever fix we put in here in the future__. So if we decide in the future to give you class name, method name and individual args as properties, we will nevertheless be stuck with this "AlternateFQN" property as legacy.

Reply:
TestWindow will use "AlternateFQN" only if it is available. If it is not available, Test Window will continue to use the FQN returned by NUnit at present.
Hence in future once we implement the right solution, we can remove this property.